streaming/index.js

```diff
@@ -622,40 +622,61 @@ const startServer = async () => {
+    const transmit = (event, raw_payload, filter_results) => {
+      let payload = raw_payload;
+      if ((event === 'update' || event === 'status.update') && filter_results && filter_results.length > 0) {
+        // copy the payload and add the `filtered` attribute, an Array of FilterResult
+        // https://docs.joinmastodon.org/entities/Status/#filtered
+        payload = {
+          ...payload,
+          filtered: filter_results
+        };
+      }
 
-      const transmit = (payload) => {
-        const now = new Date().getTime();
-        const delta = now - queued_at;
-        const encodedPayload = typeof payload === 'object' ? JSON.stringify(payload) : payload;
+      // TODO: Replace "string"-based delete payloads with object payloads:
+      const encodedPayload = typeof payload === 'object' ? JSON.stringify(payload) : payload;
 
-        log.silly(req.requestId, `Transmitting for ${accountId}: ${event} ${encodedPayload} Delay: ${delta}ms`);
-        output(event, encodedPayload);
-      };
+      log.silly(req.requestId, `Transmitting for ${accountId}: ${event} ${encodedPayload}`);
+      output(event, encodedPayload);
+    };
 
-      // Only messages that may require filtering are statuses, since notifications
-      // are already personalized and deletes do not matter
-      if (!needsFiltering || event !== 'update') {
-        transmit(payload);
+    // Currently message is of type string, soon it'll be Record<string, any>
+    const listener = message => {
+      const { event, payload } = message;
+
+      // Streaming only needs to apply filtering to some channels and only some
+      // of the events on those channels require filtering. This is because
+      // majority of the filtering happens on the Ruby on Rails (mastodon-web)
+      // side when producing the event for streaming.
+      //
+      // Specifically, the only events that require filtering are `update` and
+      // `status.update`, all other events are transmitted to the client as soon
+      // as they're received (pass-through).
+      if (!needsFiltering || (event !== 'update' && event !== 'status.update')) {
+        transmit(event, payload);
         return;
       }
 
-      const targetAccountIds = [payload.account.id].concat(payload.mentions.map(item => item.id));
-      const accountDomain = payload.account.acct.split('@')[1];
+      // The rest of the logic from here on in this function is to handle
+      // filtering of statuses
 
+      // Filter based on language:
       if (Array.isArray(req.chosenLanguages) && payload.language !== null && req.chosenLanguages.indexOf(payload.language) === -1) {
         log.silly(req.requestId, `Message ${payload.id} filtered by language (${payload.language})`);
         return;
       }
 
       // When the account is not logged in, it is not necessary to confirm the block or mute
       if (!req.accountId) {
-        transmit(payload);
+        transmit(event, payload);
         return;
       }
```

Comment:
@ClearlyClaire now that streaming is always behind authentication, we should be able to remove this check, as we should always have an `req.accountId` value (if we don't we should've failed the connection attempt)